content/automate/ManageIQ/Infrastructure/VM/Provisioning/Placement.class/__methods__/microsoft_best_fit_least_utilized.rb

```diff
@@ -1,45 +1,75 @@
 #
 # Description: This method is used to find all hosts, datastores that are the least utilized
 #
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Provisioning
+          module Placement
+            class MicrosoftBestFitLeastUtilized
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-def clear_host(prov)
-  prov.set_option(:placement_host_name, [nil, nil])
-end
+              def main
+                @handle.log("info", "vm=[#{vm.name}], space required=[#{vm.provisioned_storage}]")
+                best_fit_least_utilized
+              end
+
+              private
+
+              def request
+                @handle.root["miq_provision"].tap do |req|
+                  raise 'miq_provision not specified' if req.nil?
+                end
+              end
+
+              def vm
+                vm_obj = request.vm_template.tap do |vm|
+                  raise 'VM Object not specified' if vm.nil?
+                end
+                vm_obj.ext_management_system.tap do |ems|
+                  raise 'ext_management_system not specified' if ems.nil?
+                end
+                vm_obj
```

Comment:
How about this?
```
def vm
    request.vm_template.tap do |vm|
      raise 'VM Object not specified' if vm.nil?
      raise "EMS not found for VM [#{vm.name}]" if vm.ext_management_system.nil?
    end
end
```

Reply:
That looks really good, let me change it and test everything.  Thanks

Reply:
done